Adding a choice not to make batches in `timeseries_dataset_from_array`

europeanplaice opened the pull request:
I made a change in `timeseries_dataset_from_array` so that users can have a choice to make batches or not do.
It solves #15645.

fchollet commented on keras/preprocessing/timeseries.py:
```diff
@@ -61,7 +61,7 @@ def timeseries_dataset_from_array(
       are used for create a sample sequence.
     batch_size: Number of timeseries samples in each batch
-      (except maybe the last one).
+      (except maybe the last one). If `None`, it doesn't make batches.
```
", the data will not be batched (the dataset will yield individual samples)."

europeanplaice replied:
I changed as you suggested, Thank you!

fchollet reviewed (commented):
Thanks for the PR! This is a reasonable feature.

Please:

- Also add this in `image_dataset_from_directory` and `text_dataset_from_directory`
- Add unit tests for the feature in all 3 functions

Thank you

europeanplaice commented:
Thank you for your review. I added this in `image_dataset_from_directory` and `text_dataset_from_directory`.
Also, I added unit tests. 

The shuffle method depends on `batch_size`. If `batch_size` is None, I think `buffer_size` should be `len(dataset)` or something.

fchollet commented:
> should be len(dataset) or something.

In that case it should be a fixed constant, like 1024.

fchollet reviewed (approved):
LGTM, thank you for the contribution!

europeanplaice commented:
My pleasure.

fchollet commented:
There seems to be some failing tests -- please take a look.
